lib/src/analyzer/rules/no_material_cupertino_imports/no_material_cupertino_imports_visitor.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer_plugin/protocol/protocol_common.dart';
+import 'package:analyzer_plugin/protocol/protocol_generated.dart';
+
+import '../../rule/rule_visitor.dart';
+import '../../rule/utils.dart';
+
+class NoMaterialCupertinoImportsVisitor extends SimpleRuleVisitor {
+  NoMaterialCupertinoImportsVisitor({
+    required super.rule,
+    required super.result,
+    required super.config,
+  });
+
+  static const String material = 'package:flutter/material.dart';
+  static const String cupertino = 'package:flutter/cupertino.dart';
+  static const String widgets = 'package:flutter/widgets.dart';
+
+  @override
+  void visitImportDirective(ImportDirective node) {
+    final literal = node.uri;
+    final uri = literal.toSource();
+    final importsMaterial = uri.contains(material);
+    final importsCupertino = uri.contains(cupertino);
+    var fix = uri.replaceAll(material, widgets);
+    fix = fix.replaceAll(material, widgets);
```

Comment:
Something seems off about these statements - does it work for both `material` and `cupertino`?